packages/dialtone-vue2/components/empty_state/empty_state.vue

```diff
@@ -0,0 +1,182 @@
+      {{ bodyText }}
+    </p>
+
+    <slot name="body" />
+  </div>
+</template>
+
+<script>
+import { EMPTY_STATE_SIZE_MODIFIERS } from './empty_state_constants.js';
+import { DtIllustration, ILLUSTRATION_NAMES } from '@/components/illustration';
+import { DtIcon, ICON_NAMES } from '@/components/icon';
+
+export default {
+  name: 'DtEmptyState',
+
+  components: {
+    DtIllustration,
+    DtIcon,
+  },
+
+  props: {
+    /**
+     * The empty state size.
+     * @values 'sm', 'md', 'lg'
+     */
+    size: {
+      type: String,
+      default: 'lg',
+      validator: (s) => Object.keys(EMPTY_STATE_SIZE_MODIFIERS).includes(s),
+    },
+
+    /**
+     * The illustration name in kebab-case
+     * @type {String}
+     */
+    illustrationName: {
+      type: String,
+      default: null,
+      validator: (name) => ILLUSTRATION_NAMES.includes(name),
+    },
+
+    /**
+     * The icon name in kebab-case
+     * @type {String}
+     */
+    iconName: {
+      type: String,
+      default: null,
+      validator: (name) => ICON_NAMES.includes(name),
+    },
+
+    /**
+     * Header text
+     * @type {String}
+     */
+    headerText: {
+      type: String,
+      required: true,
+    },
+
+    /**
+     * Body text
+     * @type {String}
+     */
+    bodyText: {
+      type: String,
+      default: null,
+    },
+
+    /**
+     * Whether to show the illustration
+     * @type {Boolean}
+     */
+    showIllustration: {
+      type: Boolean,
+      default: true,
+    },
+  },
+
+  computed: {
+    /**
+     * Illustration will always be shown in lg and md size
+     * Illustration will not be shown in sm size
+     */
+    showIllustrationComponent () {
+      return this.illustrationName && this.notSmSize;
+    },
+
+    /**
+     * Icon will be shown in lg and md size only if illustration is not provided
+     * Icon will always be shown in sm size
+     */
+    showIcon () {
+      if (!this.iconName) {
+        return false;
+      }
+
+      return !(this.illustrationName && this.notSmSize);
+    },
+
+    notSmSize () {
+      return this.size !== 'sm';
+    },
+
+    sizeClass () {
+      return EMPTY_STATE_SIZE_MODIFIERS[this.size];
+    },
+
+    emptyStateClasses () {
+      return ['d-empty-state', this.sizeClass];
+    },
+  },
+
+  mounted () {
+    if (!this.bodyText && !this.$slots.body) {
+      console.warn('Dialtone Empty State component: You should provide either bodyText or content on body slot.');
+    }
+  },
+};
+</script>
+
+<style lang="scss">
+.d-empty-state {
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  justify-content: center;
+  text-align: center;
+  margin: 0 auto;
+
+  &--size-sm {
+    width: 50%;
+    h1{
+      font-size: 1.5rem;
+    }
+    p{
+      font-size: 1.2rem;
+    }
+  }
+
+  &--size-md {
+    width: 70%;
+    h1{
+      font-size: 2.7rem;
+    }
+    p{
+      font-size: 1.2rem;
+    }
+  }
+
+  &--size-lg {
+    width: 100%;
+
+    h1{
+      font-size: 3.8rem;
```

Comment:
For what I see in the designs this one should be `font: var(--dt-typography-headline-xxl)`